Tools/wasm/python.html

```diff
@@ -188,9 +221,44 @@
         this.xterm.clear();
     }
 
-    print(message) {
-        const normInput = message.replace(/[\r\n]+/g, "\n").replace(/\n/g, "\r\n");
-        this.xterm.write(normInput);
+    print(charCode) {
+        let array = [charCode];
+        if (charCode == 10) {
+            array = [13, 10];  // Replace \n with \r\n
```

Comment:
python should send \r\n not \n, line should be cooked for a terminal on stdout. add a workaround or fix tty detection in emscripten ?

Reply:
When I remove that `\n` -> `\r\n` replacement code xterm.js looks like this:

![image](https://user-images.githubusercontent.com/285352/164996647-fdee57b6-3f07-465b-a67e-d826f6fd0850.png)

When I did a `console.log` for `\r` and `\n`, I saw `\n` printed out but `\r` never seemed to come out of Python:

```python
        if (charCode == 10) {
            console.log('Saw LF');
            array = [13, 10];  // Replace \n with \r\n
        } else if (charCode == 13) {
            console.log('Saw CR');
        }
```

Reply:
it's a problem with emscripten Module I/O registration, i'm not sure but it may have to do with something like TTY.register() could not find much apart from a vague mention that it does not have any doc see https://github.com/emscripten-core/emscripten/issues/6861

Reply:
Emscripten's API is unfortunately not really a TTY and seems to be designed to work with a browser console.log for debugging than anything else. For example, it doesn't flush without a newline character. The web REPL has no choice but to work around it. 😐 

To make the Python REPL actually work nicely, you either have to do a lot more work on the JS/UI side to add behaviour (like this PR does) or add better TTY support to emscripten itself and eventually use readline (which would be awesome).

Reply:
I would also be very interested in getting `readline` to work. I think with threading enabled it would be easy. With Python in a webworker it is probably possible, but it would probably require rewriting `readline_until_enter_or_signal` and significantly patching the I/O in gnu readline itself. So maybe not completely impossible but a significant amount of work.